### PR TITLE
Make tests pass on Windows

### PR DESCRIPTION
**Pull request by crt-31**
### Description

Bunch of changes. Here's some highlights:

Code Changes
- added lineEndings=preserve to scalafmt.conf
- added .bazelrc to enable needed flags for Windows
- Refactored dependency-analyzer plugin's javaargs to use ‘;’ as a delimiter instead of ‘:’. (This made things simpler and easier since ':' as drive separator on windows and ':' is normal in a Bazel label.)
- Refactored ScalacWorker to abstract common code between Scala 2 & 3. (This consolidates some duplicated code that needed to be changed). 
- Renamed a couple function names containing rpathlocation to rlocationpath

Test Changes
- disable running tests for features not-supported on Windows
- make tests handle fact that sh_binary outputs .exe on Windows
- Handle other Windows specifics where needed (i.e. backslash as pathseparator, crlf linendings, dependence on bash)

Other notes:
- It expects git's core.autocrlf=true on windows. 

### Motivation
Keeping the tests working on Windows greatly helps ensure continuous compatibility with Windows OS.

### CODE REVIEW

File: examples/testing/multi_frameworks_toolchain/.bazelrc [hunk 1]
```diff
@@ -0,0 +1 @@
+build --enable_runfiles
```
Comment (liucijus) [inline examples/testing/multi_frameworks_toolchain/.bazelrc:1]: Can `--enable_runfiles` be put under `windows` config?

File: third_party/dependency_analyzer/src/main/io/bazel/rulesscala/dependencyanalyzer/OptionsParser.scala [hunk 2]
```diff
@@ -22,6 +22,42 @@ object OptionsParser {
...
+    def extractAndAppendToken(tokens:List[String], str:String, startIdx:Int, endIdx:Int) : List[String] = {
+      val token = str.substring(startIdx, endIdx).replace("\\", "")
+      if(!token.isEmpty()){
+        return tokens :+ token;
+      }
+      return tokens;
+    }
+
+    @annotation.tailrec
+    def tokenize(tokens:List[String], targetsStr:String, currTokenStartIdx:Int, currIdx:Int, isEscaped:Boolean) : List[String] = {
```
Comment (liucijus) [inline third_party/dependency_analyzer/src/main/io/bazel/rulesscala/dependencyanalyzer/OptionsParser.scala:38]: nit: return type probably can be `Seq[String]`

File: test/shell/test_examples.sh [hunk 3]
```diff
@@ -4,36 +4,52 @@ dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
...
 function semanticdb_example() {
-  set -e
-  ( cd examples/semanticdb; 
-    bazel build //... --aspects aspect.bzl%semanticdb_info_aspect --output_groups=json_output_file;
+
+  function build_semanticdb_example(){
+    bazel build //... --aspects aspect.bzl%semanticdb_info_aspect --output_groups=json_output_file
     bazel build //...
-  )
+    echo dude
```
Comment (liucijus) [inline test/shell/test_examples.sh:49]: nit

File: examples/testing/multi_frameworks_toolchain/.bazelrc (hunk 1 above)
Comment (crt-31) [reply examples/testing/multi_frameworks_toolchain/.bazelrc]: done.

File: third_party/dependency_analyzer/src/main/io/bazel/rulesscala/dependencyanalyzer/OptionsParser.scala (hunk 2 above)
Comment (crt-31) [reply third_party/dependency_analyzer/src/main/io/bazel/rulesscala/dependencyanalyzer/OptionsParser.scala]: done

File: test/shell/test_examples.sh (hunk 3 above)
Comment (crt-31) [reply test/shell/test_examples.sh]: haha...
Done.

**Review by simuons — APPROVED** [review]

**Pull request merged**